discopop_explorer/test/utils/validators.py

```diff
@@ -0,0 +1,40 @@
+from typing import List
+
+from discopop_explorer.pattern_detectors.do_all_detector import DoAllInfo
+
+
+def check_do_all_equivalence(do_alls_1: List[DoAllInfo], do_alls_2: List[DoAllInfo]) -> bool:
```

Comment:
I opted for a manual implementation instead since we would have to, at least, provide different comparators, depending on which types of suggestions we compare.
IMO, comparing all members of a Suggestion would / could be too pessimistic for useful testing.

Reply:
We store suggestions of different types in different lists, right? So imo there should be no problem if we test each of the lists individually.

Also we can redefine equality and sorting if we only want to test for equality of certain properties.
`def doAllInfoEqualityForTesting(d1: DoAllInfo, d2: DoAllInfo): <...>`
`DoAllInfo.__eq__ = doAllInfoEqualityForTesting`


My reasoning for the library method is that it will provide better failure messages (the current implementation only comes to a yes/no conclusion). Also it is less work and more readable to simply define what we consider as equal (in your code there is some duplication in this aspect btw, but now I am really nitpicking xD) and then let the test framework do the rest, instead of doing it ourselves. The framework is likely more optimized and more fail-safe than whatever we come up with.

e.g. I think unittest internally sorts the lists before comparing, resulting in O(n log n) vs the worst case double O(n^2) back-and-forth comparisons in our code.
Also the library ensures that multiple "equal" objects in one list have to appear just as often in the other list. In the current implementation that would not be checked.

All in all I think the unittest method wins, but then again your solution works, too. So if you want to stick to it, I am totally fine with that as well :)

Reply:
good points. You have successfully convinced me (please imagine the "achievement unlocked" sound).

Reply:
fix proposed in 1cc6e4b5243ec63cc5aa83e6246a5d350e88b1b4